lib/krb5/krbhst.c

```diff
@@ -353,13 +353,13 @@ krb5_krbhst_format_string(krb5_context context, const krb5_krbhst_info *host,
 			  char *hostname, size_t hostlen)
 {
     const char *proto = "";
-    char portstr[7] = "";
+    char portstr[7] = {0};
     if(host->proto == KRB5_KRBHST_TCP)
 	proto = "tcp/";
     else if(host->proto == KRB5_KRBHST_HTTP)
 	proto = "http://";
     if(host->port != host->def_port)
-	snprintf(portstr, sizeof(portstr), ":%d", host->port);
+	snprintf(portstr, sizeof(portstr), ":%hd", host->port);
     snprintf(hostname, hostlen, "%s%s%s", proto, host->hostname, portstr);
     return 0;
```

Comment:
I would prefer to see this rewritten to remove portstr[] entirely

  if (host->port != host->def_port)
    snprintf(hostname, hostlen, "%s%s%hd", proto, host->hostname, host->port);
  else
    snprintf(hostname, hostlen, "%s%s", proto, host->hostname);

Reply:
%hd is widely supported I presume

Reply:
I applied Jeff's fix

Reply:
That works.  `%hd` is still not needed (since it's an int), but it's fine.

Reply:
I couldn't push to @abartlet's branch to update this PR, but the fix I committed to master uses %d and an int cast.